src/Microsoft.Kiota.Abstractions.csproj

```diff
@@ -6,7 +6,7 @@
     <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
     <AssemblyTitle>Kiota Abstractions Library for dotnet</AssemblyTitle>
     <Authors>Microsoft</Authors>
-    <TargetFrameworks>netstandard2.0;netstandard2.1;</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;netstandard2.1;net5.0;net6.0;net7.0;</TargetFrameworks>
```

Comment:
even though net5.0 is a deprecated target, it's the first that supports the [trimming attributes](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-5.0). And it's still [the recommended target for libs](https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting)

What I don't know is whether we really need net6.0 and 7, from my understanding if we only added net5, any app net 5 or above would use that version.

Reply:
after further testing it seems that net 5 is all we need, resolving

Reply:
I was going to ask about target framework of net5.0 vs net6.0 since net5.0 is out of support. But it looks like the important thing is to have the trimming attributes propagated to the compiler when then the app is finally being built and the `NET5_0_OR_GREATER` flag should do the trick with the biggest target area possible.